core/src/main/java/org/mule/runtime/core/privileged/component/AnnotatedObjectInvocationHandler.java

```diff
@@ -101,13 +102,7 @@ && asList(clazz.getMethods()).stream().anyMatch(m -> "getAnnotations".equals(m.g
             .intercept(to(annotatedObjectInvocationHandler))));
     builder.set(builder.get().method(isToString().and(isDeclaredBy(Object.class))).intercept(to(new ToStringInterceptor())));
 
-    ClassLoader classLoader;
-    if (ByteBuddy.class.getClassLoader() != clazz.getClassLoader()) {
```

Comment:
We're removing this conditional, is it intentional?

Reply:
yes, that condition is already handled by BB `MultiParentClassLoader`'s builder